### PR TITLE
[FEATURE] [MER-3287] Lesson Page Outline

### PR DESCRIPTION
**Pull request by gastonabella**
[MER-3827](https://eliterate.atlassian.net/browse/MER-3827)

This PR adds a new Outline Panel on the right sidebar of the lesson view, only for practice pages.
It is not shown for graded pages or the prologue view.
The state of the Outline Panel (open or closed) is stored as a user preference in the database, so that it is reminded while navigating through different pages of the course.

https://github.com/user-attachments/assets/f3ed9843-605d-4c59-8ea2-26e7fb97bc76

[MER-3827]: https://eliterate.atlassian.net/browse/MER-3827?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/oli/accounts/schemas/user_preferences.ex [hunk 1]
```diff
@@ -6,10 +6,11 @@ defmodule Oli.Accounts.UserPreferences do
   @primary_key false
   embedded_schema do
     field :timezone, :string
+    field :page_outline_panel_active?, :boolean, default: false
```
Comment (eliknebel) [inline lib/oli/accounts/schemas/user_preferences.ex:9]: nice use of the user prefs module!

**Review by eliknebel — COMMENTED** [review]
Nice work. Separating the outline into its own live component makes a lot of sense we should consider doing it for the annotation panel as well.

File: lib/oli_web/live/delivery/student/lesson/components/outline_component.ex [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+      </defs>
+    </svg>
+    """
+  end
+
+  defp no_icon(assigns) do
+    ~H"""
+    <div role="no icon" class="flex justify-center items-center w-[22px] h-[22px] shrink-0"></div>
+    """
+  end
```
Comment (Francisco-Castro) [inline lib/oli_web/live/delivery/student/lesson/components/outline_component.ex:289]: This `no_icon` would be nice to have it in our Icons module.

**Review by Francisco-Castro — APPROVED** [review]
Nice job! 

Great use of recursion to traverse the hierarchy. Those tests make the review more enjoyable.

**Pull request merged**